opal/mca/rcache/base/rcache_base_vma_tree.c

```diff
@@ -131,6 +132,17 @@ mca_rcache_base_vma_item_t *mca_rcache_base_vma_new (mca_rcache_base_vma_module_
     return vma_item;
 }
 
+/* NTH: this function MUST not allocate or deallocate memory */
```

Comment:
I should make it clear. It is impossible (due to the lock) for the OBJ_RELEASE in that loop to be the last release on an object. I will add a comment. As for the documentation I did it through the commit message. I do no intend to let the VMA tree exist in its current form for much longer. It has serious threading performance issues.